From 4854a9e396a338920a3c8ad90e1dac9b66905251 Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Thu, 25 Aug 2016 23:13:51 -0400
Subject: [PATCH 0942/1722] drm/amd/dal: remove
 adapter_service_get_connector_obj_id and call bios directly

Change-Id: I4c731ac89ba788e5f2da503e04acf83c38e017ba
Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 .../gpu/drm/amd/dal/dc/adapter/adapter_service.c   | 26 ----------------------
 drivers/gpu/drm/amd/dal/dc/core/dc.c               |  6 ++---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c          |  5 ++---
 .../amd/dal/include/adapter_service_interface.h    |  5 -----
 4 files changed, 4 insertions(+), 38 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
index 95d1f38..a3e2afc 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
@@ -998,32 +998,6 @@ struct graphics_object_id dal_adapter_service_get_src_obj(
 	return src_object_id;
 }
 
-/** Get connector object id associated with a connector index.
- *
- * \param as	Adapter Service
- *
- * \param connector_index Index of connector between zero and total number
- *	returned by dal_adapter_service_get_connectors_num()
- *
- * \return graphics object id corresponding to the connector_index.
- */
-struct graphics_object_id dal_adapter_service_get_connector_obj_id(
-		struct adapter_service *as,
-		uint8_t connector_index)
-{
-	struct dc_bios *dcb;
-	uint8_t bios_connectors_num;
-
-	dcb = dal_adapter_service_get_bios_parser(as);
-
-	bios_connectors_num = dc_bios_get_connectors_number(dcb);
-
-	if (connector_index >= bios_connectors_num)
-		return wireless_get_connector_id(as, connector_index);
-	else
-		return dc_bios_get_connector_id(dcb, connector_index);
-}
-
 bool dal_adapter_service_get_device_tag(
 		struct adapter_service *as,
 		struct graphics_object_id connector_object_id,
diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index b7cae5c..7510d0b 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -77,13 +77,11 @@ static bool create_links(
 {
 	int i;
 	int connectors_num;
-	struct dc_bios *dcb;
+        struct dc_bios *bios = dc->ctx->dc_bios;
 
 	dc->link_count = 0;
 
-	dcb = dc->ctx->dc_bios;
-
-	connectors_num = dc_bios_get_connectors_number(dcb);
+        connectors_num = bios->funcs->get_connectors_number(bios);
 
 	if (connectors_num > ENUM_ID_COUNT) {
 		dm_error(
diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index 638fb68a..dabf8bf 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -926,6 +926,7 @@ static bool construct(
 	struct dc_context *dc_ctx = init_params->ctx;
 	struct encoder_init_data enc_init_data = { 0 };
 	struct integrated_info info = {{{ 0 }}};
+	struct dc_bios *bios = init_params->dc->ctx->dc_bios;
 
 	link->public.irq_source_hpd = DC_IRQ_SOURCE_INVALID;
 	link->public.irq_source_hpd_rx = DC_IRQ_SOURCE_INVALID;
@@ -937,9 +938,7 @@ static bool construct(
 	link->ctx = dc_ctx;
 	link->public.link_index = init_params->link_index;
 
-	link->link_id = dal_adapter_service_get_connector_obj_id(
-			as,
-			init_params->connector_index);
+	link->link_id = bios->funcs->get_connector_id(bios, init_params->connector_index);
 
 	if (link->link_id.type != OBJECT_TYPE_CONNECTOR) {
 		dm_error("%s: Invalid Connector ObjectID from Adapter Service for connector index:%d!\n",
diff --git a/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h b/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
index 432e426..1d13f89 100644
--- a/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
@@ -352,11 +352,6 @@ uint8_t dal_adapter_service_get_func_controllers_num(
 uint8_t dal_adapter_service_get_stream_engines_num(
 	struct adapter_service *as);
 
-/* functions to get object id based on object index */
-struct graphics_object_id dal_adapter_service_get_connector_obj_id(
-	struct adapter_service *as,
-	uint8_t connector_index);
-
 /* Get number of spread spectrum entries from BIOS */
 uint32_t dal_adapter_service_get_ss_info_num(
 	struct adapter_service *as,
-- 
2.7.4

